src/wallet/coin_selection.rs

```diff
@@ -579,13 +677,31 @@ impl BranchAndBoundCoinSelection {
             })
             .collect::<Vec<_>>();
 
-        BranchAndBoundCoinSelection::calculate_cs_result(selected_utxos, required_utxos, fee_amount)
+        let selected_amount = selected_utxos
+            .iter()
+            .map(|u| u.effective_value)
+            .sum::<i64>();
```

Comment:
You can probably merge this into the previous lines, replacing the `scan` with a `fold`. Keep a `(u64, Vec<_>)` as your internal state, every time you pick one utxo add the value to the accumulator and push it to the vec.

At the end you have both the selected utxos and the sum of their values

Reply:
Working on it, thanks!